discovery/pkg/openstack/reconciler.go

```diff
@@ -116,6 +119,25 @@ func (r *Reconciler) reconcile() {
 		log.Errorf("error listing OpenStack projects: %v", err)
 		return
 	}
+
+	// import white list
+	tmp := projects[:0]
+	watchedProjects := r.WatchedProjects
+	if watchedProjects != "" {
+		desireds := strings.Split(watchedProjects, ",")
+		for _, project := range projects {
+			for _, desired := range desireds {
+				if desired == project.Name {
+					tmp = append(tmp, project)
+				}
+			}
+		}
+		projects = tmp
```

Comment:
This will cause problems if the user provides a project that doesn't exist.

Reply:
I think I would have taken a slightly different approach here. Instead of overwriting the projects we got from the OpenStack API, I would have added a check in the for loop (line 138):

```
for _, project := range projects {
  // skip the project if it is not in the whitelist
  if !contains(whitelist, project) { // pseudo-code here... this checks whether the project is in the whitelist
    continue
  }
  ...
  ...
```

Reply:
@alexbrand I applied your approach. Please check my commit.